internal/datanode/compaction_executor.go

```diff
@@ -34,10 +35,10 @@ const (
 )
 
 type compactionExecutor struct {
```

Comment:
also move compactionExecutor to datanode/compaction/ dir?

Reply:
Too many changes in one PR, maybe you could move it inside.